.github/workflows/release.yml

```diff
@@ -49,6 +49,24 @@ jobs:
         id: changesets
         with:
           publish: yarn changeset publish
+          version: yarn changeset:version:custom
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Publish Solidity Library
+        if: steps.changesets.outputs.published == 'true'
+        uses: actions/github-script@v5
+        env:
+          PUBLISHED_PACKAGES: ${{ steps.changesets.outputs.publishedPackages }}
+        with:
+          script: |
+            const publishedPackages = JSON.parse(process.env.PUBLISHED_PACKAGES);
+            const contractsPackage = publishedPackages.find((pkg) => pkg.name === '@sphinx-labs/contracts');
+            if (contractsPackage) {
+              github.rest.git.createRef({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                ref: `refs/tags/v${contractsPackage.version}`,
+                sha: context.sha
+              });
+            }
```

Comment:
Creates a new GitHub tag in the format `vx.y.z` whenever a new release of the contracts package happens. This is somewhat difficult to fully test locally, but I'm quite confident it will work as I have tested the JS script that creates the tag.